src/ui/popup.js

```diff
@@ -482,12 +490,18 @@ export default class Popup extends Evented {
      *
      * @example
      * let popup = new mapboxgl.Popup()
-     * popup.toggleClassName('toggleClass')
+     * popup.toggleClassName('highlighted')
      */
     toggleClassName(className: string) {
         if (this._container) {
-            return this._container.classList.toggle(className);
+            this._container.classList.toggle(className);
+        }
+        if (this._classList.has(className)) {
```

Comment:
I think you can directly use `delete` here because it returns `true` if the entry existed in the set and was successfully removed.

Reply:
Can you elaborate on what you mean by that @arindam1993?
Are you suggesting changing the following line to:
```js
return this.classList.delete(className);
```

Reply:
No, i meant 
```
if(this.classList.delete(className)){
   finalState = false; 
} else {
   finalState = true;
   this._classList.add(className);
}
```

I think `Set#delete` already internally does a `Set#has` check, so I was suggesting leveraging  that

Reply:
Good idea!